hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncTableImpl.java

```diff
@@ -822,7 +826,25 @@ private <S, R> void onLocateComplete(Function<RpcChannel, S> stubMaker,
         } else {
           callback.onRegionComplete(region, r);
         }
-        if (unfinishedRequest.decrementAndGet() == 0 && locateFinished.get()) {
+        ServiceCaller<S, R> updatedCallable = callback.getNextCallable(r, region);
```

Comment:
This logic is as suggested by @bbeaudreault on the mailing list https://lists.apache.org/thread/1vqnxb71z7swq2cogz4qg3cn6b10xp4v